### DISCUSSION TITLE
Change resolution during stream

### DISCUSSION THREAD

**Discussion by MishaMgla** [HaishinKit/HaishinKit.swift#1473] [Q&A]
### Describe the bug

Changing resolution to the other than was initially declared during streaming encoded incorrectly (huge glitches).
I found in issues that I should set profileLevel and sessionPreset, however it still not working.
If settings use as setup stream settings, it works. It fails only if I change to video settings other than was initially declared
```
func updateResolution(_ resolution: String) {
        if (resolution=="1080p") {
            stream?.videoSettings.profileLevel = "kVTProfileLevel_H264_High_AutoLevel"
            stream?.sessionPreset = AVCaptureSession.Preset.hd1920x1080
            stream?.videoSettings.videoSize.width = 1920;
            stream?.videoSettings.videoSize.height = 1080;
}}
````

### To Reproduce

1. Setup stream
2. Start stream
3. Change video settings to resolution other than was declared in stream setup

### Expected behavior

Able to change resolution on the fly

### Version

1.8.0

### Smartphone info.

Device: IPhone 13 pro max

### Screenshots

<img width="637" alt="image" src="https://github.com/shogo4405/HaishinKit.swift/assets/98478673/e891c7a1-c140-4854-850d-5f23b1b8c82c">

**Comment by shogo4405** [comment]
Changing the resolution during streaming requires support from both the server and the viewing player. In many cases, making changes during streaming can cause issues and is generally not recommended.

To reproduce the issue on my end, could you please provide the names of the server and viewing player products that encountered the problem?

**Comment by MishaMgla** [comment]
> Changing the resolution during streaming requires support from both the server and the viewing player. In many cases, making changes during streaming can cause issues and is generally not recommended.
> 
> To reproduce the issue on my end, could you please provide the names of the server and viewing player products that encountered the problem?

Sure, it is Ant Media Server with default settings and embedded player in AMS admin panel.

**Comment by shogo4405** [comment]
I see. It might be possible to watch without the green screen issue using tools like ffmpeg or ffplay. I recommend checking how it works in your actual viewing environment. To avoid any trouble, it is best to temporarily stop and then resume the stream.

**Comment by MishaMgla** [comment]
> I see. It might be possible to watch without the green screen issue using tools like ffmpeg or ffplay. I recommend checking how it works in your actual viewing environment. To avoid any trouble, it is best to temporarily stop and then resume the stream.

Ty, I`ll use stop and resume approach

**Comment by shogo4405** [comment]
Changing the resolution during streaming often causes issues, which could be due to the middleware, HaishinKit, or the viewing player. Since we cannot definitively say it is a problem with HaishinKit, I will move this to the Discussion section.

If anyone has expertise in this area, please let me know.
